src/charm.py

```diff
@@ -9,76 +9,181 @@
+        elif self.topic != new_topic:
+            self.unit.status = ActiveStatus(
+                f"New database name specified: {new_database_name}. Please remove current relation/s and add a new relation!"
+            )
+            return
+
+        if self.extra_user_roles == "":
+            self.extra_user_roles == new_extra_user_roles
+        elif self.extra_user_roles != new_extra_user_roles:
+            self.unit.status = ActiveStatus(
+                f"New user-extra-roles specified: {new_extra_user_roles}. Please remove current relation/s and add a new relation!"
+            )
+            return
 
+        # kafka
         self.mysql.database = self.database
         self.postgresql.database = self.database
         self.mongodb.database = self.database
-
+        self.kafka.topic = self.topic
+
+        database_relation_data = {
+            "database": self.database,
+            "extra-user-roles": self.extra_user_roles,
+        }
+
+        self.update_database_relations(database_relation_data)
+
+        # look if kafka is connected
+        for rel in self.kafka.relations:
+            self.kafka._update_relation_data(
+                rel.id,
+                {
+                    "topic": self.topic,
+                    "extra-user-roles": KAFKA_EXTRA_USER_ROLES + self.extra_user_roles,
+                },
+            )
+
+        # output the correct message.
+        self.set_status(event)
+
+    def update_database_relations(self, database_relation_data: Dict):
+        """Update the relation data of the related databases."""
         for rel in self.mysql.relations:
-            self.mysql._update_relation_data(rel.id, {"database": self.database})
+            self.mysql._update_relation_data(rel.id, database_relation_data)
         for rel in self.postgresql.relations:
-            self.postgresql._update_relation_data(rel.id, {"database": self.database})
+            self.postgresql._update_relation_data(rel.id, database_relation_data)
         for rel in self.mongodb.relations:
-            self.mongodb._update_relation_data(rel.id, {"database": self.database})
+            self.mongodb._update_relation_data(rel.id, database_relation_data)
+
+    def set_status(self, event):
+        """Set the unit status."""
+        if self.database == "" and self.topic == "":
+            self.unit.status = BlockedStatus("The database name or topic name is not specified.")
+        elif self.database == "":
+            if self.is_kafka_related:
+                self.unit.status = ActiveStatus(f"The topic name is specified: {self.topic}.")
+            else:
+                self.unit.status = ActiveStatus(
+                    f"The topic name is specified: {self.topic}. Add relation with Kakfa operator."
+                )
+        elif self.topic == "":
+            if self.is_database_related:
+                self.unit.status = ActiveStatus(
+                    f"The database name is specified: {self.database}."
```

Comment:
I believe the convention is to NOT add messages for ActiveStatus, if it is all good. But please @marcoppenheimer , double check this is correct.